File: app_manager/appmanager/src/main/java/app/editors/manager/ui/views/media/MediaVideoView.java
package app.editors.manager.ui.views.media;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.VideoView;

public class MediaVideoView extends VideoView {

    public interface OnMediaVideoListener {
        void onMediaStart();
        void onMediaPause();
        void onMediaSuspend();
        void onMediaResume();
    }

    protected OnMediaVideoListener mOnMediaVideoListener;

    public MediaVideoView(Context context) {
        super(context);
    }

    public MediaVideoView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MediaVideoView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public MediaVideoView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    public void start() {
        super.start();
        if (mOnMediaVideoListener != null) {
            mOnMediaVideoListener.onMediaStart();
        }
    }

    @Override
    public void pause() {
        super.pause();
        if (mOnMediaVideoListener != null) {
            mOnMediaVideoListener.onMediaPause();
        }
    }

    @Override
    public void suspend() {
        super.suspend();
        if (mOnMediaVideoListener != null) {
            mOnMediaVideoListener.onMediaSuspend();
        }
    }

    @Override
    public void resume() {
        super.resume();
        if (mOnMediaVideoListener != null) {
            mOnMediaVideoListener.onMediaResume();
        }
    }

    public void setOnMediaVideoListener(final OnMediaVideoListener onMediaVideoListener) {
        mOnMediaVideoListener = onMediaVideoListener;
    }

}

File: toolkit/libtoolkit/src/main/java/lib/toolkit/base/ui/views/media/MediaVideoView.kt
package lib.toolkit.base.ui.views.media

import android.content.Context
import android.util.AttributeSet
import android.widget.VideoView

class MediaVideoView : VideoView {

    protected var mOnMediaVideoListener: OnMediaVideoListener? = null

    interface OnMediaVideoListener {
        fun onMediaStart()
        fun onMediaPause()
        fun onMediaSuspend()
        fun onMediaResume()
    }

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    override fun start() {
        super.start()
        mOnMediaVideoListener?.onMediaStart()
    }

    override fun pause() {
        super.pause()
        mOnMediaVideoListener?.onMediaPause()
    }

    override fun suspend() {
        super.suspend()
        mOnMediaVideoListener?.onMediaSuspend()
    }

    override fun resume() {
        super.resume()
        mOnMediaVideoListener?.onMediaResume()
    }

    fun setOnMediaVideoListener(onMediaVideoListener: OnMediaVideoListener) {
        mOnMediaVideoListener = onMediaVideoListener
    }

}
